packages/minimongo/cursor.js

```diff
@@ -379,7 +361,14 @@ export default class Cursor {
 
     // run the observe callbacks resulting from the initial contents
     // before we leave the observe.
-    await this.collection._observeQueue.drain();
+    const isReadyPromise = this.collection._observeQueue.drain();
+
+    if (Meteor.isClient) handle.isReady = true;
```

Comment:
These Meteor.isClient checks should probably instead check what type of minimongo collection/cursor it is. Unnamed collections and collections created for DDP clients should work the same way even if they are on the server.